src/fs_sys_err.h

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+/* These will all be returned in syscalls which have a return type of unit32_t
+ * This means that they will all be very high constants because of two's
+ * complement on low absolute value negatives being nearest to the maximum
+ * value of 2^32. Hopefully this is far out of conflict with bytes_read
+*/
+
+#define ERR_FD_OOR -1   // file descriptor is out of range
```

Comment:
So in syscall.h I return an int32_t for each of open/read/write/close, but the function syscall( ) itself returns a uint32_t, so anything below that might as well return a uint32_t as well.